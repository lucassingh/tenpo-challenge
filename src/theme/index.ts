import light from './lightTheme';

export default { light }