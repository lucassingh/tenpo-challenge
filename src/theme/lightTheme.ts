export default {
    colors: {
        primary: "#D4F9F5",
        secondary: "#00BAA4",
        background: "#F2F2F2",
        lightGray: "#F2F2F2",
        light: "#FFFFFF",
        dark: "#1C1C1C"
    }
}