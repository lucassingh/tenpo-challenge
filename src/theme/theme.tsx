import React from "react";
import { ThemeProvider } from "styled-components/native";
import theme from ".";

type Props = {
    children: React.ReactNode
}

export function Theme({children}: Props) {
    return <ThemeProvider theme={theme.light}>
        {children}
    </ThemeProvider>
}