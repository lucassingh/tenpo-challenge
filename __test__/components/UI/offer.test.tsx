import React from "react";
import 'jest-styled-components'
import styled from 'styled-components/native';
import theme from "../../../src/theme";
import { render } from '@testing-library/react-native'

describe('Test OfferContainer and TextOffer in component in <Offer />', () => {

    const OfferContainer = styled.View`
        display: flex;
        width: 30px;
        height: 30px;
        padding: 5px;
        border-radius: 20px;
        display: flex;
        justify-content: center;
        align-items: center;
        background: ${theme.light.colors.secondary};
        position: absolute;
        right: 7px;
        top:-10px;
        z-index: 10;    
    `

    it('Should match with snapshot', () => {

        const container = render(<OfferContainer/>);

        expect(container).toMatchSnapshot()
    })

    const TextOffer = styled.Text`
        color: ${theme.light.colors.light}
        font-size: 7px;
    `

    it('Should match with snapshot', () => {

        const container = render(<TextOffer />);

        expect(container).toMatchSnapshot()
    })
})