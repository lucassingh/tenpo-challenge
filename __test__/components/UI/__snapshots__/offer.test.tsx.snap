// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OfferContainer and TextOffer in component in <Offer /> Should match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#00BAA4",
        "borderRadius": 20,
        "display": "flex",
        "height": 30,
        "justifyContent": "center",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
        "position": "absolute",
        "right": 7,
        "top": -10,
        "width": 30,
        "zIndex": 10,
      },
    ]
  }
/>
`;

exports[`Test OfferContainer and TextOffer in component in <Offer /> Should match with snapshot 2`] = `
<Text
  style={
    [
      {
        "color": "#FFFFFF",
        "fontSize": 7,
      },
    ]
  }
/>
`;
